lib/src/rules/deprecated_consistency.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Missing deprecated annotation.';
+
+const _details = r'''
+
+Do apply `@Deprecated()` consistently:
+
+- if a class is deprecated its contructors should also be deprecated.
+- if a field is deprecated the contructor parameter pointing to it should also be deprecated.
```

Comment:
=> if a field is deprecated, the constructor parameter pointing to it should also be deprecated.